Linux: Bulk Export PDF Dialog Without Permissions

### Operating system

Linux

### Joplin version

3.0.13

### Desktop version info

Joplin 3.0.13 (prod, linux)

客户端 ID: fbc564639de9413998f0449ecaf31d0e
同步版本：3
配置文件版本：47
支持的密钥链：否

修订：599cf5b

Backup: 1.4.1

### Current behaviour

![Image](https://github.com/user-attachments/assets/f24e9472-5ca1-4631-a2c6-56f95ce1a22b)
![Image](https://github.com/user-attachments/assets/5fcdb374-e07e-42e6-b216-7a2c222c9388)
![Image](https://github.com/user-attachments/assets/9c52464d-1e81-4f88-8541-29c15c3007f6)
![Image](https://github.com/user-attachments/assets/18ff18c9-c0dd-4e00-bee6-cd4be8335100)
1. Use Ctrl to select multiple notes, then click the export button in the updated interface on the right
2. Select PDF Export
3. Do not select any location, just click cancel (the position where the dialog opens does not affect the reappearance of this bug)
4. Multiple dialogs without root directory read permissions will appear, and they will repeat based on the number of selected documents.

### Expected behaviour

Expected Joplin to stop exporting after I deselected the location, instead of continuing to attempt exporting after the path was deselected

### Logs

```log
MainScreen.tsx:466 Error: Error: EACCES: permission denied, open '/2024***1.pdf'. Path: /2024******1.pdf
    at FsDriverNode.fsErrorToJsError_ (/tmp/.mount_Joplin3NwXBi/resources/app.asar/node_modules/@joplin/lib/fs-driver-node.js:23:24)
    at FsDriverNode.<anonymous> (/tmp/.mount_Joplin3NwXBi/resources/app.asar/node_modules/@joplin/lib/fs-driver-node.js:52:28)
    at Generator.throw (<anonymous>)
    at rejected (/tmp/.mount_Joplin3NwXBi/resources/app.asar/node_modules/@joplin/lib/fs-driver-node.js:6:65)
```

![图片](https://github.com/user-attachments/assets/025139fe-6cb4-4375-a593-5a33dc3031e6)

--- a/packages/app-desktop/gui/MainScreen/commands/exportPdf.ts
+++ b/packages/app-desktop/gui/MainScreen/commands/exportPdf.ts
@@ -3,7 +3,7 @@ import shim from '@joplin/lib/shim';
 import InteropServiceHelper from '../../../InteropServiceHelper';
 import { _ } from '@joplin/lib/locale';
 import Note from '@joplin/lib/models/Note';
-const bridge = require('@electron/remote').require('./bridge').default;
+import bridge from '../../../services/bridge';
 
 export const declaration: CommandDeclaration = {
 	name: 'exportPdf',
@@ -29,6 +29,14 @@ export const runtime = (comp: any): CommandRuntime => {
 					path = await bridge().showOpenDialog({
 						properties: ['openDirectory', 'createDirectory'],
 					});
+				}
+
+				if (Array.isArray(path)) {
+					if (path.length > 1) {
+						throw new Error('Only one output directory can be selected');
+					}
+
+					path = path[0];
 				}
 
 				if (!path) return;

